homeassistant/components/hunterdouglas_powerview/cover.py

```diff
@@ -136,21 +169,51 @@ def __init__(self, coordinator, device_info, room_name, shade, name):
         self._is_opening = False
         self._is_closing = False
         self._last_action_timestamp = 0
+        self._power_connection = 0
+        if isinstance(self._shade.raw_data[SHADE_BATTERY_KIND], int):
+            self._power_connection = self._shade.raw_data[SHADE_BATTERY_KIND]
         self._scheduled_transition_update = None
-        self._current_hd_cover_position = MIN_POSITION
+        self._current_hd_cover_primary = MIN_POSITION
+        self._current_hd_cover_secondary = MIN_POSITION
         if self._device_info[DEVICE_MODEL] != LEGACY_DEVICE_MODEL:
             self._attr_supported_features |= CoverEntityFeature.STOP
         self._forced_resync = None
 
     @property
     def extra_state_attributes(self):
         """Return the state attributes."""
-        return {STATE_ATTRIBUTE_ROOM_NAME: self._room_name}
+        return {
+            STATE_ATTRIBUTE_ROOM_NAME: self._room_name,
+            STATE_ATTRIBUTE_POWER_CONN: SHADE_POWER_CONN_LIST[self._power_connection],
+        }
+
+    @property
+    def get_position_primary(self):
+        """Access shade position directly from co-ordinator."""
+        if self.coordinator.data is None:
+            return self._current_hd_cover_primary
```

Comment:
Let's remove the `data is None` guard since the next PR will take care of ensuring its never `None`

Reply:
see https://github.com/home-assistant/core/pull/62788#discussion_r868014354